mws/sitesmanagement/views/others.py

```diff
@@ -64,6 +64,12 @@ def clone_vm_view(request, site_id):
     if site is None:
         return HttpResponseForbidden()
 
+    can_upgrade = False
+    if settings.MAX_PENDING_UPGRADES:
+        pending_upgrades = len([x for x in Service.objects.filter(type='test') if x.active])
```

Comment:
I'll add to the queryset .prefetch_related('virtual_machines') as this they are used by .active to check if the Service is active, thus reducing the number of queries to the database to only one instead of 1 per Service.

```
Service.objects.filter(type='test').prefetch_related('virtual_machines')
```